pkg/controller/submariner/submariner_controller.go

```diff
@@ -204,7 +204,7 @@ func newEngineDaemonSet(cr *submarinerv1alpha1.Submariner) *appsv1.DaemonSet {
 		ObjectMeta: metav1.ObjectMeta{
 			Labels:    labels,
 			Namespace: cr.Namespace,
-			Name:      "submariner",
+			Name:      "submariner-gateway",
```

Comment:
UTs need to be updated too. Perhaps we should make this a constant?

Reply:
Oh, I had a look on UTs and didn't see a check on the name.

It's better not to use a constant for UTs to reference. That provides double validation (if you make a mistake in one of the sides, or change the constant without noticing, the UTs will chime at you)